multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -138,10 +141,57 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 				}
 			}
 		}
-		// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-		// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-		// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
 	}
+	// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
```

Comment:
Why we move the error out of the "if" block?

err here should be returned by r.List() at line 131? If so, I do not feel it is relevant to the comment. @luolanzone : could you check?

Reply:
@jianjuns   yes, you are right. I feel original comment and logic doesn't handle the case correctly. If all ClusterClaims are deleted, it should be empty list instead of an error. I think it should retry if there is an error. @hjiajing Could you double check and refine this part? we may leave ClusterClaim webhook part in a new PR.